### ISSUE TITLE
Update the support-matrix

### ISSUE THREAD

**Issue by Madhu-1** [ceph/ceph-csi#4852]
https://github.com/ceph/ceph-csi?tab=readme-ov-file#support-matrix is pretty old not not updated, we need to update all the supported features to GA, we usually to from alpha to Beta, and Beta to GA as this was not well maintained, IMO we can directly instead most of the status to GA (except topology)

@nixpanic @Rakshith-R thoughts?

**Comment by nixpanic** [comment]
Yes, move them all to GA. Unless there are known issues or major limitations, those should be documented in an issue and linked from the documentation so that interested users can contribute to improving the feature.
